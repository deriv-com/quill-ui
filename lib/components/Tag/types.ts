import { ReactNode } from "react";
import { TRegularSizesWithExtraSmall } from "@types";

export type BaseTagProps = {
    variant?: "fill" | "outline" | "custom";
    color?: "error" | "success" | "custom" | "warning" | "info";
    isBold?: boolean;
    icon?: ReactNode;
    iconClassName?: string;
    size?: TRegularSizesWithExtraSmall;
    label?: ReactNode;
    className?: string;
};
