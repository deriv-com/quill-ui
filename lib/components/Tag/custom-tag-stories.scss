
.custom-tag{
    background: linear-gradient(90deg, var(--core-color-solid-blue-100), var(--core-color-solid-blue-300), var(--core-color-solid-blue-500));
    color: var(--core-color-solid-blue-1200);
    border: 1px solid var(--core-color-solid-blue-700);
    &-icon {
       fill: var(--core-color-solid-blue-1200)
    }
    &-without-icon-fill {
        background-color: var(--core-color-solid-yellow-100);
        color: var(--core-color-solid-yellow-1200);
    }
    &-without-icon-outline {
        border: 1px solid var(--core-color-solid-red-1200);
        color: var(--core-color-solid-red-700)!important;
        &-icon {
            fill: var(--core-color-solid-red-1200)
        }
    }
}

