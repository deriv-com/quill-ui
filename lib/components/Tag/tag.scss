.tag {
    min-width: auto;
    border-radius: var(--core-borderRadius-200);
    display: inline-flex;
    align-items: center;
    justify-self: center;

    &__size {
        &-sm {
            padding: 3px 8px;
            gap: 8px;
        }

        &-xs {
            padding: 2px 4px;
            gap: 4px
        }

        &-md {
            padding: 5px 12px;
            gap: 12px;
        }

        &-lg {
            padding: 12px 16px;
            gap: 16px;
        }
    }

    &__color {
        &--success {
            color: var(--core-color-solid-green-800);

            &-fill {
                background-color: var(--core-color-opacity-green-100);
            }

            &-outline {
                border: 1px solid var(--core-color-solid-green-700);
            }

            &-svg {
                fill: var(--core-color-solid-green-800)
            }
        }

        &--error {
            color: var(--core-color-solid-red-800);

            &-fill {
                background-color: var(--core-color-opacity-red-100);
            }

            &-outline {
                border: 1px solid var(--core-color-solid-red-700);
            }

            &-svg {
                fill: var(--core-color-solid-red-800)
            }
        }

        &--warning {
            color: var(--core-color-solid-orange-800);

            &-fill {
                background-color: var(--core-color-opacity-orange-100);
            }

            &-outline {
                border: 1px solid var(--core-color-solid-orange-700);
            }

            &-svg {
                fill: var(--core-color-solid-orange-800)
            }
        }

        &--info {
            color: var(--core-color-solid-blue-800);

            &-fill {
                background-color: var(--core-color-opacity-blue-100);
            }

            &-outline {
                border: 1px solid var(--core-color-solid-blue-700);
            }

            &-svg {
                fill: var(--core-color-solid-blue-800)
            }
        }
    }
}