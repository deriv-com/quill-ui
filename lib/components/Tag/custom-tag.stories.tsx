import type { Meta, StoryObj } from "@storybook/react";
import {
    LabelPairedPlaceholderSmRegularIcon,
    LabelPairedThumbsUpCaptionFillIcon,
} from "@deriv/quill-icons/LabelPaired";
import "./custom-tag-stories.scss";
import Tag from ".";

const meta = {
    title: "Components/Tag/Custom",
    component: Tag,
    parameters: {
        layout: "centered",
    },
    args: {
        label: "Custom",
        size: "md",
        isBold: true,
        variant: "custom",
    },
    argTypes: {
        label: {
            name: "Label",
            control: "text",
        },
        icon: {
            table: {
                disable: true,
            },
        },
        size: {
            options: ["xs", "sm", "md", "lg"],
            control: { type: "radio" },
            description: "To select the size of icon and font-size",
        },

        className: {
            name: "Class Name",
            control: "text",
        },
        iconClassName: {
            table: {
                disable: true,
            },
        },
        isBold: {
            control: "boolean",
        },
        color: {
            table: {
                disable: true,
            },
        },
    },
    tags: ["autodocs"],
} satisfies Meta<typeof Tag>;

export default meta;
type Story = StoryObj<typeof meta>;

export const CustomTagFillWithIcon: Story = {
    args: {
        variant: "custom",
        icon: <LabelPairedPlaceholderSmRegularIcon />,
        className: "custom-tag",
        iconClassName: "custom-tag-icon",
    },
};
export const CustomTagWithIcon: Story = {
    args: {
        variant: "custom",
        icon: <LabelPairedThumbsUpCaptionFillIcon />,
        className: "custom-tag-outline",
        label: "Best seller",
        iconClassName: "custom-tag-outline-icon",
        size: "lg",
    },
};

export const CustomTagWithoutIcon: Story = {
    args: {
        variant: "custom",
        className: "custom-tag-without-icon-outline",
        isBold: false,
        size: "md",
    },
};
