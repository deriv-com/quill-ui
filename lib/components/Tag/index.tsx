import { forwardRef } from "react";
import { BaseTagProps } from "./types";
import clsx from "clsx";
import { CaptionText, Text } from "@components/Typography";
import "./tag.scss";
import TagIcon from "./tagIcon";

export const Tag = forwardRef<HTMLDivElement, BaseTagProps>(
    (
        {
            label,
            className,
            variant = "fill",
            size = "md",
            color = "success",
            isBold = false,
            icon,
            iconClassName,
            showIcon = true,
            ...rest
        },
        ref,
    ) => {
        const tagColor = variant === "custom" ? "custom" : color;
        return (
            <div
                className={clsx(
                    className,
                    "tag",
                    `tag__size-${size}`,
                    `tag__color--${tagColor}`,
                    `tag__color--${tagColor}-${variant}`,
                )}
                style={
                    variant === "custom"
                        ? { backgroundColor: color }
                        : undefined
                }
                {...rest}
                ref={ref}
            >
                {showIcon && (
                    <TagIcon
                        isBold={isBold}
                        className={iconClassName}
                        icon={icon}
                        color={tagColor}
                        size={size}
                    />
                )}

                {size === "sm" || size === "xs" ? (
                    <CaptionText bold={isBold} color={tagColor}>
                        {label}
                    </CaptionText>
                ) : (
                    <Text
                        size={size === "md" ? "sm" : "md"}
                        bold={isBold}
                        color={tagColor}
                    >
                        {label}
                    </Text>
                )}
            </div>
        );
    },
);

Tag.displayName = "Tag";

export default Tag;
