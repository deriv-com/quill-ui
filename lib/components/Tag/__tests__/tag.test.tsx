import { render } from "@testing-library/react";
import { StandalonePlaceholderRegularIcon } from "@deriv/quill-icons/Standalone";
import Tag from "../index";

describe("Tag", () => {
    it("Should render Tag Default", () => {
        const { container } = render(<Tag label="label"></Tag>);
        expect(container).toMatchSnapshot();
    });

    it("Should render Custom with Icon", () => {
        const { container } = render(<Tag label="label"></Tag>);
        expect(container).toMatchSnapshot();
    });

    it("Should render Custom Tag with Icon", () => {
        const { container } = render(
            <Tag
                icon={StandalonePlaceholderRegularIcon}
                variant="custom"
                color="custom"
                label="Label"
            ></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Custom Tag without Icon", () => {
        const { container } = render(
            <Tag variant="custom" color="custom" label="Label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Custom Tag Fill without Icon", () => {
        const { container } = render(
            <Tag color="custom" variant="fill" label="Label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Custom Tag Outline without Icon", () => {
        const { container } = render(
            <Tag color="custom" variant="outline" label="Label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Custom Tag Outline with Icon", () => {
        const { container } = render(
            <Tag
                icon={StandalonePlaceholderRegularIcon}
                color="custom"
                variant="outline"
                label="Label"
            ></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Error Tag Fill", () => {
        const { container } = render(<Tag color="error" label="label"></Tag>);
        expect(container).toMatchSnapshot();
    });

    it("Should render Warning Tag Fill", () => {
        const { container } = render(<Tag color="warning" label="label"></Tag>);
        expect(container).toMatchSnapshot();
    });

    it("Should render Success Tag Fill", () => {
        const { container } = render(
            <Tag color="success" label="laabel"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Info Tag Fill", () => {
        const { container } = render(<Tag color="info" label="label"></Tag>);
        expect(container).toMatchSnapshot();
    });

    it("Should render Error Tag Outline", () => {
        const { container } = render(
            <Tag color="error" variant="outline" label="label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Warning Tag Outline", () => {
        const { container } = render(
            <Tag color="warning" variant="outline" label="label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Success Tag Outline", () => {
        const { container } = render(
            <Tag color="success" variant="outline" label="label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });

    it("Should render Info Tag Outline", () => {
        const { container } = render(
            <Tag color="info" variant="outline" label="label"></Tag>,
        );
        expect(container).toMatchSnapshot();
    });
});
