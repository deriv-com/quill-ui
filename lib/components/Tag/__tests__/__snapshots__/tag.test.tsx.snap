// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Should render Custom Tag Fill without Icon 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--custom tag__color--custom-fill"
  >
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default custom"
    >
      Label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Custom Tag Outline with Icon 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--custom tag__color--custom-outline"
  >
    <div
      height="22"
      width="14"
    >
      <svg
        height="32"
        role="img"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.688 7.75h.937c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625h-.937C9.477 9 8.5 10.016 8.5 11.188v.937a.64.64 0 0 1-.625.625.617.617 0 0 1-.625-.625v-.937a3.443 3.443 0 0 1 3.438-3.438M7.874 14c.313 0 .625.313.625.625v3.75a.64.64 0 0 1-.625.625.617.617 0 0 1-.625-.625v-3.75c0-.312.273-.625.625-.625m16.25 0c.313 0 .625.313.625.625v3.75a.64.64 0 0 1-.625.625.617.617 0 0 1-.625-.625v-3.75c0-.312.273-.625.625-.625m0-1.25a.617.617 0 0 1-.625-.625v-.937C23.5 10.016 22.484 9 21.313 9h-.938a.617.617 0 0 1-.625-.625c0-.312.273-.625.625-.625h.938c1.875 0 3.437 1.563 3.437 3.438v.937a.64.64 0 0 1-.625.625m.625 8.125v.938a3.443 3.443 0 0 1-3.437 3.437h-.938a.617.617 0 0 1-.625-.625c0-.312.273-.625.625-.625h.938c1.171 0 2.187-.977 2.187-2.187v-.938c0-.312.273-.625.625-.625.313 0 .625.313.625.625m-16.25 0v.938c0 1.21.977 2.187 2.188 2.187h.937c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625h-.937a3.42 3.42 0 0 1-3.438-3.437v-.938c0-.312.273-.625.625-.625.313 0 .625.313.625.625m5.625 4.375a.617.617 0 0 1-.625-.625c0-.312.273-.625.625-.625h3.75c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625zM13.5 8.375c0-.312.273-.625.625-.625h3.75c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625h-3.75a.617.617 0 0 1-.625-.625"
        />
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default custom"
    >
      Label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Custom Tag Outline without Icon 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--custom tag__color--custom-outline"
  >
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default custom"
    >
      Label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Custom Tag with Icon 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--custom tag__color--custom-custom"
  >
    <div
      height="22"
      width="14"
    >
      <svg
        height="32"
        role="img"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.688 7.75h.937c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625h-.937C9.477 9 8.5 10.016 8.5 11.188v.937a.64.64 0 0 1-.625.625.617.617 0 0 1-.625-.625v-.937a3.443 3.443 0 0 1 3.438-3.438M7.874 14c.313 0 .625.313.625.625v3.75a.64.64 0 0 1-.625.625.617.617 0 0 1-.625-.625v-3.75c0-.312.273-.625.625-.625m16.25 0c.313 0 .625.313.625.625v3.75a.64.64 0 0 1-.625.625.617.617 0 0 1-.625-.625v-3.75c0-.312.273-.625.625-.625m0-1.25a.617.617 0 0 1-.625-.625v-.937C23.5 10.016 22.484 9 21.313 9h-.938a.617.617 0 0 1-.625-.625c0-.312.273-.625.625-.625h.938c1.875 0 3.437 1.563 3.437 3.438v.937a.64.64 0 0 1-.625.625m.625 8.125v.938a3.443 3.443 0 0 1-3.437 3.437h-.938a.617.617 0 0 1-.625-.625c0-.312.273-.625.625-.625h.938c1.171 0 2.187-.977 2.187-2.187v-.938c0-.312.273-.625.625-.625.313 0 .625.313.625.625m-16.25 0v.938c0 1.21.977 2.187 2.188 2.187h.937c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625h-.937a3.42 3.42 0 0 1-3.438-3.437v-.938c0-.312.273-.625.625-.625.313 0 .625.313.625.625m5.625 4.375a.617.617 0 0 1-.625-.625c0-.312.273-.625.625-.625h3.75c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625zM13.5 8.375c0-.312.273-.625.625-.625h3.75c.313 0 .625.313.625.625a.64.64 0 0 1-.625.625h-3.75a.617.617 0 0 1-.625-.625"
        />
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default custom"
    >
      Label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Custom Tag without Icon 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--custom tag__color--custom-custom"
  >
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default custom"
    >
      Label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Custom with Icon 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--success tag__color--success-fill"
  >
    <div
      class="tag__color--success-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m2.926-8.422-3.5 3.5a.463.463 0 0 1-.63 0l-1.75-1.75a.463.463 0 0 1 0-.629.463.463 0 0 1 .63 0l1.449 1.45 3.172-3.2a.463.463 0 0 1 .629 0 .463.463 0 0 1 0 .63"
          />
        </g>
        <defs>
          <clippath
            id="ab98f299d0d18e1e48b223c656245044__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default success"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Error Tag Fill 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--error tag__color--error-fill"
  >
    <div
      class="tag__color--error-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M.93 15.262c-.055.082-.055.164-.055.246 0 .273.219.52.492.52h11.238c.274 0 .52-.247.52-.52a.44.44 0 0 0-.082-.246L7.547 5.828A.66.66 0 0 0 7 5.5a.67.67 0 0 0-.574.328zm-.766-.438L5.66 5.391A1.57 1.57 0 0 1 7 4.625c.52 0 1.04.3 1.313.766l5.496 9.433c.109.192.191.438.191.684 0 .765-.629 1.367-1.395 1.367H1.367A1.354 1.354 0 0 1 0 15.508c0-.246.055-.492.164-.684M7 8.125a.47.47 0 0 1 .438.438v3.5A.45.45 0 0 1 7 12.5a.43.43 0 0 1-.437-.437v-3.5A.45.45 0 0 1 7 8.124m-.656 6.125c0-.355.273-.656.656-.656.355 0 .656.3.656.656 0 .383-.3.656-.656.656a.63.63 0 0 1-.656-.656"
          />
        </g>
        <defs>
          <clippath
            id="3bda4ccac60989735a89bdc8e6b254bd__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default error"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Error Tag Outline 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--error tag__color--error-outline"
  >
    <div
      class="tag__color--error-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M.93 15.262c-.055.082-.055.164-.055.246 0 .273.219.52.492.52h11.238c.274 0 .52-.247.52-.52a.44.44 0 0 0-.082-.246L7.547 5.828A.66.66 0 0 0 7 5.5a.67.67 0 0 0-.574.328zm-.766-.438L5.66 5.391A1.57 1.57 0 0 1 7 4.625c.52 0 1.04.3 1.313.766l5.496 9.433c.109.192.191.438.191.684 0 .765-.629 1.367-1.395 1.367H1.367A1.354 1.354 0 0 1 0 15.508c0-.246.055-.492.164-.684M7 8.125a.47.47 0 0 1 .438.438v3.5A.45.45 0 0 1 7 12.5a.43.43 0 0 1-.437-.437v-3.5A.45.45 0 0 1 7 8.124m-.656 6.125c0-.355.273-.656.656-.656.355 0 .656.3.656.656 0 .383-.3.656-.656.656a.63.63 0 0 1-.656-.656"
          />
        </g>
        <defs>
          <clippath
            id="3bda4ccac60989735a89bdc8e6b254bd__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default error"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Info Tag Fill 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--info tag__color--info-fill"
  >
    <div
      class="tag__color--info-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m-1.312-4.375h.875V10.75h-.657a.43.43 0 0 1-.437-.437.45.45 0 0 1 .437-.438H7a.47.47 0 0 1 .438.438v3.062h.875a.47.47 0 0 1 .437.438.45.45 0 0 1-.437.437H5.687a.43.43 0 0 1-.437-.437.45.45 0 0 1 .438-.438M7 8.781a.63.63 0 0 1-.656-.656c0-.355.273-.656.656-.656.355 0 .656.3.656.656 0 .383-.3.656-.656.656"
          />
        </g>
        <defs>
          <clippath
            id="1c46262142574a7a3675d7c9e5e6dfb8__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default info"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Info Tag Outline 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--info tag__color--info-outline"
  >
    <div
      class="tag__color--info-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m-1.312-4.375h.875V10.75h-.657a.43.43 0 0 1-.437-.437.45.45 0 0 1 .437-.438H7a.47.47 0 0 1 .438.438v3.062h.875a.47.47 0 0 1 .437.438.45.45 0 0 1-.437.437H5.687a.43.43 0 0 1-.437-.437.45.45 0 0 1 .438-.438M7 8.781a.63.63 0 0 1-.656-.656c0-.355.273-.656.656-.656.355 0 .656.3.656.656 0 .383-.3.656-.656.656"
          />
        </g>
        <defs>
          <clippath
            id="1c46262142574a7a3675d7c9e5e6dfb8__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default info"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Success Tag Fill 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--success tag__color--success-fill"
  >
    <div
      class="tag__color--success-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m2.926-8.422-3.5 3.5a.463.463 0 0 1-.63 0l-1.75-1.75a.463.463 0 0 1 0-.629.463.463 0 0 1 .63 0l1.449 1.45 3.172-3.2a.463.463 0 0 1 .629 0 .463.463 0 0 1 0 .63"
          />
        </g>
        <defs>
          <clippath
            id="ab98f299d0d18e1e48b223c656245044__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default success"
    >
      laabel
    </p>
  </div>
</div>
`;

exports[`Tag Should render Success Tag Outline 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--success tag__color--success-outline"
  >
    <div
      class="tag__color--success-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m2.926-8.422-3.5 3.5a.463.463 0 0 1-.63 0l-1.75-1.75a.463.463 0 0 1 0-.629.463.463 0 0 1 .63 0l1.449 1.45 3.172-3.2a.463.463 0 0 1 .629 0 .463.463 0 0 1 0 .63"
          />
        </g>
        <defs>
          <clippath
            id="ab98f299d0d18e1e48b223c656245044__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default success"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Tag Default 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--success tag__color--success-fill"
  >
    <div
      class="tag__color--success-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m2.926-8.422-3.5 3.5a.463.463 0 0 1-.63 0l-1.75-1.75a.463.463 0 0 1 0-.629.463.463 0 0 1 .63 0l1.449 1.45 3.172-3.2a.463.463 0 0 1 .629 0 .463.463 0 0 1 0 .63"
          />
        </g>
        <defs>
          <clippath
            id="ab98f299d0d18e1e48b223c656245044__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default success"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Warning Tag Fill 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--warning tag__color--warning-fill"
  >
    <div
      class="tag__color--warning-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m0-10.5a.47.47 0 0 1 .438.438v3.5a.45.45 0 0 1-.438.437.43.43 0 0 1-.437-.437v-3.5A.45.45 0 0 1 7 7.25m.656 6.125c0 .383-.3.656-.656.656a.63.63 0 0 1-.656-.656c0-.355.273-.656.656-.656.355 0 .656.3.656.656"
          />
        </g>
        <defs>
          <clippath
            id="2158e3ea37743710bbd193f7fe99b23d__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default warning"
    >
      label
    </p>
  </div>
</div>
`;

exports[`Tag Should render Warning Tag Outline 1`] = `
<div>
  <div
    class="tag tag__size-md tag__color--warning tag__color--warning-outline"
  >
    <div
      class="tag__color--warning-svg"
      height="22"
      width="14"
    >
      <svg
        height="22"
        role="img"
        viewBox="0 0 14 22"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M7 4.625a6.11 6.11 0 0 0-5.305 3.063c-1.12 1.914-1.12 4.238 0 6.125C2.79 15.727 4.785 16.875 7 16.875a6.04 6.04 0 0 0 5.277-3.062c1.121-1.887 1.121-4.211 0-6.126A6.1 6.1 0 0 0 7 4.626M7 17.75a6.96 6.96 0 0 1-6.07-3.5c-1.258-2.16-1.258-4.812 0-7A7.02 7.02 0 0 1 7 3.75c2.488 0 4.785 1.34 6.043 3.5 1.258 2.188 1.258 4.84 0 7A6.95 6.95 0 0 1 7 17.75m0-10.5a.47.47 0 0 1 .438.438v3.5a.45.45 0 0 1-.438.437.43.43 0 0 1-.437-.437v-3.5A.45.45 0 0 1 7 7.25m.656 6.125c0 .383-.3.656-.656.656a.63.63 0 0 1-.656-.656c0-.355.273-.656.656-.656.355 0 .656.3.656.656"
          />
        </g>
        <defs>
          <clippath
            id="2158e3ea37743710bbd193f7fe99b23d__a"
          >
            <path
              d="M0 0h14v22H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <p
      class="quill-typography__body-text__size--sm__weight--regular__decoration--default warning"
    >
      label
    </p>
  </div>
</div>
`;
